app/gui/src/controller/searcher/component.rs

```diff
@@ -291,9 +302,24 @@ impl List {
     /// All groups from [`List`] without the groups found in [`List::favorites`].
     fn all_groups_not_in_favorites(&self) -> impl Iterator<Item = &Group> {
         let normal = self.module_groups.values().map(|mg| &mg.content);
-        let flattened = self.top_modules_flattened.iter();
+        let flattened = self.top_module_sections.iter().flat_map(|s| s.modules_flattened.iter());
         normal.chain(flattened).chain(std::iter::once(&self.local_scope))
     }
+
+    /// Get a vector of section names for the sections of the top modules.
+    pub fn top_module_section_names(&self) -> Vec<&ImString> {
+        self.top_module_sections.iter().map(|s| &s.name).collect()
+    }
```

Comment:
Again, I'd return the `impl Iterator`. If someone needs to collect it to `Vec`, they may do it themselves.